### PR TITLE
 Fix media removal cancel upload

### LINKED ISSUE #627: Media upload spinner still displays after media is removed
When uploading media, if you try to delete a video/image by backspacing over it, the nav bar "uploading" progress bar still displays: https://cloudup.com/cBmDmgY8RAD

I am assuming the media is actually still being uploaded, so we should also consider if the upload should be killed (if possible).

Ref: https://github.com/wordpress-mobile/WordPress-iOS/pull/3503

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #627 

How to test:

Insert a media object on the post (video or image), check if on removal of the media event while the upload is happening a console message say "Media Removed: XXX"

Needs Review: @astralbodies

### CODE REVIEW

**Comment by astralbodies** [comment]
Reviewing now @SergioEstevao :smile:

File: Assets/ZSSRichTextEditor.js [hunk 1]
```diff
@@ -2361,16 +2376,21 @@ ZSSField.prototype.bindMutationObserver = function () {
...
-                                                          }
-                                                          });
-                                        });
+        mutations.forEach(function(mutation) {
+            var removedNode = ZSSEditor.searchForMediaNode(mutation.removedNodes)
+            if ( removedNode != null) {                
+                var mediaIdentifier = ZSSEditor.extractMediaIdentifier(removedNode);
+                ZSSEditor.sendMediaRemovedCallback(mediaIdentifier);
+            }
+//            for (var i = 0; i < mutation.removedNodes.length; i++) {
```
Comment (astralbodies) [inline Assets/ZSSRichTextEditor.js]: Commented code - maybe should be deleted?

File: Example/EditorDemo/WPViewController.m [hunk 2]
```diff
@@ -149,6 +149,7 @@ - (void)editorViewController:(WPEditorViewController *)editorViewController medi
 {
     NSProgress * progress = self.mediaAdded[mediaID];
     [progress cancel];
+    NSLog(@"Media Removed: %@", mediaID);
```
Comment (astralbodies) [inline Example/EditorDemo/WPViewController.m]: Its weird I'm not seeing these log messages in EditorDemo.

**Comment by astralbodies** [comment]
It looks like `-editorViewController:mediaRemoved:` is only being called when I have a post with just an image and I either delete it or click on it to cancel. If I put an image between two paragraphs of text this method is not called.

File: Example/EditorDemo/WPViewController.m (hunk 2 above)
Comment (astralbodies) [inline Example/EditorDemo/WPViewController.m]: Should probably use DDLog i/o NSLog

**Comment by SergioEstevao** [comment]
@astralbodies ready for another round of review. I converted the NSLog to DDLogInfo so remember to switch to the correct log level on the WPEditorDemoLoggingConfiguration.m in order to see the remove message.

**Comment by astralbodies** [comment]
Looks like deleting the image calls the delegate - awesome! It appears that tapping the image and selecting cancel upload does not call the delegate now ever.

**Comment by SergioEstevao** [comment]
@astralbodies: This is expected because when tapping and selecting to
cancel then code path goes the other way around, the Native code sends a
message to the web view to remove the image, so no callback is needed to
notify that the image was removed.

**Comment by astralbodies** [comment]
Got it - then we're good :shipit:

**Comment by astralbodies** [comment]
As long as @sendhil  is okay with putting this in 6.1 :smile:

**Comment by SergioEstevao** [comment]
This is a bad bug to have in editor so I should we should add it to 6.1.

**Comment by SergioEstevao** [comment]
I'm going to merge this on the editor, then we can make the decision on the main repo if we want this fix in.

**Pull request merged**